rust/numaflow-core/src/shared/server_info.rs

```diff
@@ -351,7 +351,12 @@ async fn read_server_info(
     // Infinite loop to keep checking until the file is ready
     loop {
         if cln_token.is_cancelled() {
-            return Err(Error::ServerInfo("Operation cancelled".to_string()));
+            return Err(Error::ServerInfo(format!(
+                "Server info file {:?} is not ready. \
+                This indicates that the server is not started correctly in the user-code. \
+            Eg (simple sink in Java, similarly you can refer language specific SDK examples) https://github.com/numaproj/numaflow-java/blob/2aeaafff1c6dec7fd66e018b142ef6fe4ffcf0a9/examples/src/main/java/io/numaproj/numaflow/examples/sink/simple/SimpleSink.java#L23",
```

Comment:
Another reason you might see this error is if the server is not up because the app code is still running the `initialization` code (e.g, downloading some cache files, etc). 

IMO, just write this
* This indicates that the server has not started
* Provide a link to an FAQ in the numalfow github page and in that page we can add more details.

@KeranYang WDYT?

Reply:
@vigith +1.